controllers/drpolicy_controller.go

```diff
@@ -378,5 +387,67 @@ func (u *objectUpdater) finalizerRemove() error {
 func (r *DRPolicyReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ramen.DRPolicy{}).
+		Watches(
+			&source.Kind{Type: &corev1.ConfigMap{}},
+			handler.EnqueueRequestsFromMapFunc(r.configMapMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
+		Watches(
+			&source.Kind{Type: &corev1.Secret{}},
+			handler.EnqueueRequestsFromMapFunc(r.secretMapFunc),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
```

Comment:
(future) what if a secret was deleted and then recreated (to say correct the values), we possibly need to watch for those as well.

Reply:
Good point.  I suppose an issue is needed to track support watching for create, delete, and general? events.

Reply:
Secret creates and deletes are now being watched in addition to updates.  The scenario you described was added to the envtest test suite.